### ISSUE TITLE
Locally Connected Layer

### ISSUE THREAD

**Issue by lolz0r** [torch/nn#322]
Hello,
Is it possible to have a layer which is like a fully connected layer but instead of each node being fully the nodes are only partly (locally) connected to the next layer?

**Comment by soumith** [comment]
You can do this with several parallel nn.Linear layers, using a combination of nn.Select / nn.Narrow and nn.Linear contained in an nn.Concat I think.

**Comment by boknilev** [comment]
Hi @lolz0r , 
were you able to implement a partly-connected layer this way?

**Comment by lolz0r** [comment]
Hi boknilev,
I found an out of the box implementation inside the lib fbnn so I just used that one.

On Aug 26, 2015 5:14 AM, boknilev <notifications@github.com> wrote:
Hi @lolz0r ,
were you able to implement a partly-connected layer this way?

---

**Comment by boknilev** [comment]
That one looks like a locally connected convolution (https://github.com/facebook/fbnn/blob/master/fbnn/LocallyConnected.lua), right? Or is it the same as a non-convolutional locally-connected layer?

**Comment by lolz0r** [comment]
I thought it was the same (?) lol

**Comment by jblan016** [comment]
I've been trying to implement the last conv layer in this paper for the last 5 days (http://liris.cnrs.fr/Documents/Liris-6107.pdf), that is "N1".
I used Soumith's advice. And I was able to train my neural net with the following model:
model = nn.Sequential()
c = nn.Concat(2)
model:add(nn.SpatialConvolutionMM(1,32,3,3,1,1)) --I(1x20x20)->O(32x18x18)
--model:add(nn.Dropout())
model:add(nn.SpatialMaxPooling(3,3,2,2,1,1)) --O(32x9x9)
model:add(nn.ReLU())
model:add(nn.SpatialConvolutionMM(32,32,3,3,1,1))--O(32x7x7)
--model:add(nn.Dropout())
model:add(nn.ReLU())
model:add(nn.SpatialMaxPooling(3,3,2,2,0,0)) --O(32x3x3)
--model:add(nn.SpatialConvolutionMM(32,32,3,3,1,1))--O(32x1x1)
for i=1,32 do
  local t = nn.Sequential()
  t:add(nn.Select(2,i))
  t:add(nn.Unsqueeze(2))
  t:add(nn.SpatialConvolutionMM(1,1,3,3,1,1))
  t:add(nn.ReLU())
  t:add(nn.Reshape(1))
  c:add(t)
end
model:add(c)
model:add(nn.Linear(32,2))
model:add(nn.LogSoftMax())

However when I try to use this to create a heatmap with nn.SpatialClassifier on an image pyramid I get an error. I believe that my error is related to the batch mode. I do not get any error if I don't do this fully connected layer in parallel.
